### PR TITLE
Change hiera_array to the newer lookup

### LINKED ISSUE #983: function hiera_array is deprecated
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.17.0
- Ruby: 2.0.0
- Distribution: CentOS Linux release 7.8.2003
- Module version: 2.0.0

## Output log
Puppet The function 'hiera_array' is deprecated in favor of using 'lookup'. See https://puppet.com/docs/puppet/6.17/deprecated_language.html

https://github.com/voxpupuli/puppet-jenkins/blob/c0eaac41e0e3218b5f832ba0980c372948bb03c8/manifests/slave.pp#L163
https://github.com/voxpupuli/puppet-jenkins/blob/c0eaac41e0e3218b5f832ba0980c372948bb03c8/manifests/slave.pp#L173
https://github.com/voxpupuli/puppet-jenkins/blob/c0eaac41e0e3218b5f832ba0980c372948bb03c8/manifests/slave.pp#L183

### PR DESCRIPTION
**Pull request by drzombey**
#### Pull Request (PR) description
This PR remove the deprecated hiera_array and change it to the new lookup.

#### This Pull Request (PR) fixes the following issues
Fixes #983

### CODE REVIEW

File: manifests/slave.pp [hunk 1]
```diff
@@ -160,7 +160,7 @@
 
   if $labels {
     if $labels =~ Array {
-      $_combined_labels = hiera_array('jenkins::slave::labels', $labels)
+      $_combined_labels = lookup( { 'name' => 'jenkins::slave::labels', 'default_value' => $labels } )
```
Comment (ekohl) [inline manifests/slave.pp:163]: I wonder if this is really needed. The hiera key `jenkins::slave::labels` means `$labels` already contains the correct content, right? I think that you can simply drop `hiera_array` everywhere.

Comment (alexjfisher) [reply manifests/slave.pp:163]: Not quite.  Specifically, `hiera_array` would have been merging arrays across the hierarchy.

Comment (alexjfisher) [reply manifests/slave.pp:163]: You need to either set the merge behaviour as a parameter to `lookup` here, or set lookup_options data in module hiera.  See https://github.com/voxpupuli/puppet-telegraf/pull/134 for example.

**Review by alexjfisher — CHANGES_REQUESTED** [review]

File: manifests/slave.pp (hunk 1 above)
Comment (drzombey) [reply manifests/slave.pp:163]: You mean something like this `lookup('jenkins::slave::labels', Any, unique, $labels)` ?

Comment (alexjfisher) [reply manifests/slave.pp:163]: yeah, something like is needed to preserve behaviour.  Bonus points if you can rid of it and use `lookup_options` in module hiera though.  That is more flexible as users can override the merge behaviour in their own environment hiera if they fancy.

But don't worry if you can't get that working.  Switching to `lookup()` would still be an improvement.

**Comment by ekohl** [comment]
Replaced by https://github.com/voxpupuli/puppet-jenkins/pull/1055

**Pull request closed without merging**